src/replication.c

```
@@ -2634,6 +3387,11 @@ void syncWithPrimary(connection *conn) {
             if (err) goto write_error;
         }
 
+        /* Announce that the replica is capable of dual channel sync. */
+        if (server.dual_channel_replication) {
+            err = sendCommand(conn, "REPLCONF", "capa", "dual-channel", NULL);
+        }
```

This err is never checked.

Missing `if (err) goto write_error;`?

---

Btw, this doesn't matter, but I believe it should be possible to send this together with all the other capas. The primary will ignore capabilities it does not understand.

---

Right, I'll add a check

---

> Btw, this doesn't matter, but I believe it should be possible to send this together with all the other capas. The primary will ignore capabilities it does not understand.

You're right, we can indeed combine this with the other capabilities now. We've postponed the primary capability check until after the `+DUALCHANNELSYNC` reply, which makes this approach feasible.